File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/alertas_stock.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

[Table("alertas_stock")]
public partial class alertas_stock
{
    [Key]
    public int id_alerta { get; set; }

    [Required(ErrorMessage = "El id del producto es obligatorio.")]
    public int id_producto { get; set; }

    [Required(ErrorMessage = "El nivel mínimo es obligatorio.")]
    [Range(0, int.MaxValue, ErrorMessage = "El nivel mínimo debe ser un valor positivo.")]
    public int nivel_minimo { get; set; }

    [Required(ErrorMessage = "La fecha de la alerta es obligatoria.")]
    [Column(TypeName = "datetime")]
    [DataType(DataType.DateTime, ErrorMessage = "Formato de fecha inválido.")]
    public DateTime fecha_alerta { get; set; }

    [ForeignKey("id_producto")]
    [InverseProperty("alertas_stocks")]
    public virtual producto id_productoNavigation { get; set; } = null!;
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/alertasStock.js
const mongoose = require('mongoose');
const { Schema } = mongoose;
const Producto = require('../models/producto');  // Asegúrate de ajustar la ruta según tu estructura


const alertasStockSchema = new Schema({
    nivel_minimo: {
        type: Number,
        required: [true, 'El nivel mínimo es obligatorio.'],
        min: [0, 'El nivel mínimo debe ser un valor positivo.']
    },
    fecha_alerta: {
        type: Date,
        required: [true, 'La fecha de la alerta es obligatoria.'],
        
    },
    id_productoNavigation: {
        type: Schema.Types.ObjectId,
        ref: 'Producto' // Asegúrate de que este sea el nombre correcto del modelo producto
    }
});

// Exportar el modelo
module.exports = mongoose.model('AlertasStock', alertasStockSchema);
